src/services.py

```diff
@@ -143,3 +158,66 @@ def __init__(self, Security=Security):
     def get_all(self):
         with session_scope() as session:
             return [sec.asdict() for sec in session.query(self.Security).all()]
+
+
+class LinkedinService:
+    def __init__(self, User=User, UserService=UserService):
+        self.User = User
+        self.UserService = UserService
+
+    @validate_input(LINKEDIN_BUYER_PRIVILEGES_SCHEMA)
+    def activate_buyer_privileges(self, code, redirect_uri, user_email):
+        user_data = self.get_user_data(code=code, redirect_uri=redirect_uri)
+        is_email = self.is_email_matched(
+            linkedin_email=user_data.get("email"), user_email=user_email
+        )
+        if is_email:
+            user = self.UserService().get_user_by_email(email=user_email)
+            return self.UserService().activate_buy_privileges(user_id=user.get("id"))
+        else:
+            raise InvalidRequestException("Linkedin email does not match")
+
+    @validate_input(LINKEDIN_CODE_SCHEMA)
```

Comment:
If it is not going to be called with the API I think we do not need to validate the params. My main intention for validating the params is that so that it can be returned to the API directly if there are validation mismatch. But if it's an internal tool I think not necessary.